test/python/taskbasedsimulator/benchmark.py

```diff
@@ -154,11 +154,11 @@ def run_benchmark(
     qc.remove_final_measurements(inplace=True)
     qc = transpile(qc, basis_gates=basis_gates_transpile, optimization_level=0)
     for i in range(qc.size()):
-        if qc[i][0].name == "u2" or qc[i][0].name == "u3":
+        if qc[i][0].name == "u2" or qc[i][0].name == "u3":  # noqa: PLR1714
```

Comment:
Why not change these occurrences too?
It's more performant and reads almost the same.

Reply:
To keep this block consistent between the equality checks and the `.startswith` checks. I think in this particular case, the difference in performance will be negligible.

Reply:
I agree that it does not really matter that much, but I do feel that the more consistent and performant way would just be to change all occurrences and replace the `mcx` condition by something of the form
```python
if qc[i][0].name in ('cx', 'ccx') or qc[i][0].name.startswith(...)
```
That way it is consistent and performant.

Reply:
You got me on consistent. But I'd still argue that performance is a non-issue in this section of the code ;)